src/o365/graph/commands/reports/reports-teams-device-usage-user-detail.ts

```diff
@@ -0,0 +1,153 @@
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.period = typeof args.options.period !== 'undefined';
+    telemetryProps.date = typeof args.options.date !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    const periodParameter: string = args.options.period ? `getTeamsDeviceUsageUserDetail(period='${encodeURIComponent(args.options.period)}')` : '';
+    const dateParameter: string = args.options.date ? `getTeamsDeviceUsageUserDetail(date=${encodeURIComponent(args.options.date)})` : '';
+
+    let endpoint: string = args.options.period ? `${auth.service.resource}/v1.0/reports/${periodParameter}` : `${auth.service.resource}/v1.0/reports/${dateParameter}`;
+
+    this.getUsageData(endpoint, cmd)
+      .then((res): void => {
+        cmd.log(res);
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, cmd, cb));
+
+  }
+
+  protected getUsageData(url: string, cmd: CommandInstance): Promise<void> {
+    return new Promise<void>((resolve: (data: any) => void, reject: (error: any) => void): void => {
+      auth
+        .ensureAccessToken(auth.service.resource, cmd, this.debug)
+        .then((): Promise<any> => {
+          const requestOptions: any = {
+            url: url,
+            headers: {
+              authorization: `Bearer ${auth.service.accessToken}`
+            }
+          };
+
+          return request.get(requestOptions);
+        })
+        .then((res: any): void => {
+          resolve(res);
+        }, (err: any): void => {
+          reject(err);
+        });
+    });
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-p, --period [period]',
+        description: 'Specify the length of time over which the report is aggregated. The supported values are D7|D30|D90|D180',
+        autocomplete: ['D7', 'D30', 'D90', 'D180']
+      },
+      {
+        option: '-d, --date [date]',
+        description: 'Specify the date for which you would like to view the users who performed any activity. The supported date format is YYYY-MM-DD'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(): CommandValidate {
+
+    return (args: CommandArgs): boolean | string => {
+
+      if (args.options.period && args.options.date) {
+        return 'You can\'t use period and date parameters together.';
+      }
+
+      if (args.options.period) {
+        const period: string = args.options.period.toUpperCase();
+        if (period !== 'D7' && period !== 'D30' && period !== 'D90' && period !== 'D180') {
+          return `${period} is not a valid period type. The supported values are D7|D30|D90|D180`;
+        }
+      }
+
+      const parsedDate = Date.parse(args.options.date as string);
```

Comment:
Unfortunately `Date.parse` is too broad as it allows to using other date formats such as `yyyy/MM/dd` as well, which would parse just fine but which would fail in the MS Graph.

Reply:
Hi Waldek, 
Referred the following command but please let me know the code change required, I will make changes as per your feedback. This is the only point pending in this PR feedbacks. 

If it is a simple fix, please feel free to change the code and process the PR. 

https://raw.githubusercontent.com/pnp/office365-cli/c13ba0388ebc1579a6ae727783350bb4466e3cd1/src/o365/spo/commands/list/list-webhook-add.ts

Reply:
Maybe a regex to validate the fornat of the string?  
args.options.date.match(/^\d{4}-\d{2}-\d{2}$/)

Reply:
@DaveCalGH, Thank you for your suggestion. Let us wait for @waldekmastykarz's response and change accordingly.

Reply:
As @DaveCalGH suggested, it would be better to adjust the validation to validate the specific date format.

Reply:
Hi @waldekmastykarz, I have fixed the date issue as suggested by Dave. 

Also removed the 30 days validation as the API documentation(says 30 days) and the error I get from the graph API(Only support data for the past 28 days.) are conflicting each other. Let us fix this later after clarifying with Microsoft.  

Please proceed with PR.